tensornetwork/ncon_interface.py

```diff
@@ -17,19 +17,23 @@
 from __future__ import division
 from __future__ import print_function
 import warnings
-from typing import Any, Sequence, List, Optional, Union, Text, Tuple, Dict
+from typing import Any, Sequence, List, Optional, Union, Text, Tuple, Dict, Set
 from tensornetwork import network
 from tensornetwork import network_components
-
+# pylint: disable=useless-import-alias
+import tensornetwork.config as config
+# pylint: disable=useless-import-alias
+import tensornetwork.backends.backend_factory as backend_factory
 Tensor = Any
 
 
-def ncon(tensors: Sequence[Tensor],
+def ncon(tensors: Sequence[Union[network_components.BaseNode, Tensor]],
```

Comment:
I still don't see the value in having `BaseNode`s as an input to `ncon`

Reply:
it just feels more streamlined. With `Node` our API provides a wrapper for `Tensor` so that we can do backend independent tensor contractions. The way I use it is to build my code around `Node` objects, not `Tensor` (apart from initialization). So it seems nice not to have to access `Node.tensor` when calling `ncon`.